### ISSUE TITLE
Failed: CompatFormFieldSpec

### ISSUE THREAD

**Issue by chbatey** [akka/akka-http#2612]
https://jenkins.akka.io:8498/job/akka-http-nightly/559/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.13.0,jdk=AdoptOpenJDK%2011/console

```
[info] CompatFormFieldSpec:
[info] FormFieldDirectives
[info]   should be compatible
[info] akka.http.scaladsl.server.directives.CompatFormFieldSpec *** ABORTED ***
[info]   java.lang.NoSuchMethodError: akka.http.scaladsl.server.Directives$.formField(Lakka/http/scaladsl/server/directives/FormFieldDirectives$FieldMagnet;)Ljava/lang/Object;
[info]   at akka.http.scaladsl.server.directives.CompatFormField$.oneParameter(CompatFormField.scala:12)
[info]   at akka.http.scaladsl.server.directives.CompatFormFieldSpec.$anonfun$new$3(CompatFormFieldSpec.scala:17)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$3.apply(WordSpecLike.scala:1075)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[info]   ...
```

**Comment by jrudolph** [comment]
This is a real incompatibility for 2.13.0 (because I introduced new signatures using `@pre213` / `@since213`). Do we care enough to fix those in the next release, wdyt @raboof?

**Comment by raboof** [comment]
> Do we care enough to fix those in the next release

Hmm, if we fix it in the next release 10.1.9 will not be compatible with 10.1.10, though... and we're using `formField` in akka-management.

Is it realistically possible to fix this in the next release, without just rolling the whole thing back?

I'm leaning towards keeping the changes, and updating/releasing akka-management to use at least 10.1.9. As 2.13 is still quite new, likely users that update to it have no problems also updating to the latest akka-management. Does that make sense?

**Comment by jrudolph** [comment]
> Is it realistically possible to fix this in the next release, without just rolling the whole thing back?

Theoretically, yes, because the fix for 2.11/2.12 would work for 2.13 as well but since we hadn't released a version for 2.13 yet back then, I opted for evolving to a more reasonable result with 2.13. Then in the meantime 10.1.8 for 2.13 was released and we ended up in this situation.

The lessons learned would be
 * to be careful with diverging code between Scala versions because test coverage may not uncover all problems (soon enough)
 * if we want to quickly release a version for a new Scala version, maybe flag the first version as preliminary, so that there's some additional time for considerations afterwards to prepare the first full release.

> I'm leaning towards keeping the changes, and updating/releasing akka-management to use at least 10.1.9. As 2.13 is still quite new, likely users that update to it have no problems also updating to the latest akka-management. Does that make sense?

I think it makes sense. How bad is it for akka-management? We have to update and force everyone on 2.13 to upgrade to the new version? That's probably still reasonable.
